src/main/java/com/internet/shop/controllers/order/CompleteOrderController.java

```diff
@@ -0,0 +1,27 @@
+package com.internet.shop.controllers.order;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/complete")
+public class CompleteOrderController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        orderService.completeOrder(shoppingCartService.getByUserId(USER_ID));
```

Comment:
According to the checklist, we should separate calling via service into the local variable. still, it has no impact on the work :)

Reply:
That point is concerning situation when you call service method that fetcher data from db, for example  shoppingCartService.getByUserId() several times. It's better to save result into local variable in such case.